### ISSUE TITLE
"brew cask install joinme" fails on Catalina

### ISSUE THREAD

**Issue by core-code** [Homebrew/homebrew-cask#71226]
it seems there is a bug in the 'Catalina' version of 'ditto' so that it can't extract some ZIP files. the cask 'joinme' is affected:

```
brew cask install joinme
==> Satisfying dependencies
==> Downloading https://secure.join.me/Download.aspx?installer=mac&webdownload=true
######################################################################## 100.0%
==> No SHA-256 checksum defined for Cask 'joinme', skipping verification.
==> Installing Cask joinme
cp: /var/folders/4s/clr73zjj3fdgh6dbd47yx49m0000gn/T/d20191021-47293-1jc5p99/__MACOSX/.: Permission denied
==> Purging files for version latest of Cask joinme
Error: Failure while executing; `cp -pR /var/folders/4s/clr73zjj3fdgh6dbd47yx49m0000gn/T/d20191021-47293-1jc5p99/__MACOSX/. /usr/local/Caskroom/joinme/latest/__MACOSX` exited with 1. Here's the output:
cp: /var/folders/4s/clr73zjj3fdgh6dbd47yx49m0000gn/T/d20191021-47293-1jc5p99/__MACOSX/.: Permission denied

Follow the instructions here:
  https://github.com/Homebrew/homebrew-cask#reporting-bugs
/usr/local/Homebrew/Library/Homebrew/system_command.rb:176:in `assert_success!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:53:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:29:in `run'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:33:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:19:in `system_command!'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy/directory.rb:21:in `block in extract_to_dir'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy/directory.rb:20:in `each'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy/directory.rb:20:in `extract_to_dir'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy.rb:123:in `extract'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy.rb:143:in `block in extract_nestedly'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
/usr/local/Homebrew/Library/Homebrew/unpack_strategy.rb:127:in `extract_nestedly'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:187:in `extract_primary_container'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:77:in `stage'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:99:in `install'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:22:in `block in run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:16:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:16:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/abstract_command.rb:36:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:92:in `run_command'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:158:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:123:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:9:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:102:in `<main>'

```

i am going to send a bug report about 'ditto' to Apple now (not that they would ever fix anything)

**Comment by core-code** [comment]
http://openradar.appspot.com/radar?id=4944708392452096

**Comment by vitorgalvao** [comment]
Was the `bug.zip` you attached renamed from joinme or did you recreate the error? Asking so I can make a duplicate report (allegedly that helps prioritise issues).

**Comment by core-code** [comment]
yes it was just a renamed download from 'joinme'.

i am not sure how to create a ZIP file that reproduces the problem myself.

**Comment by nuhkoca** [comment]
Hi @core-code 

As discussed on this issue #70511  Catalina needs your terminal to have **Full Disk Access**

To enable it;

Please go to System Preferences, tab "Privacy" and allow total disk access to Terminal.

**Comment by core-code** [comment]
@nuhkoca there is no connection between the mentioned bug in ‘ditto’ and the resulting inability to install ‘joinme’ and the ‘full disk access’

**Comment by nuhkoca** [comment]
@core-code I see, sorry. I thought it is related when I saw **Permission denied** in the log.

**Comment by core-code** [comment]
no problem ;-)

**Comment by core-code** [comment]
this seems fixed in macOS 10.15.2
